testsuite/features/step_definitions/command_steps.rb

```diff
@@ -956,6 +956,12 @@
   end
 end
 
+When(/^I delete all "([^"]*)" volumes from "([^"]*)" pool on "([^"]*)" without error control$/) do |volumes, pool, host|
+  node = get_target(host)
+  output, _code = node.run("virsh vol-list #{pool} | sed -n -e 's/^[[:space:]]*\([^[:space:]]\+\).*$/\1/;/#{volumes}/p'", false)
+  output.each_line { |volume| node.run("virsh vol-delete #{volume} #{pool}", false) }
+end
```

Comment:
Yes, we should have similar steps instead of the commands `I run`. The cucumber text should remain high-level and hide the gory technical details of implementation.